### PR TITLE
Use ServiceBinding types instead of unstructured.Unstructured

### LINKED ISSUE #4690: Use `ServiceBinding` types instead of `unstructured.Unstructured`
To work around issues with a new  [service binding operator](https://github.com/redhat-developer/service-binding-operator) version we had to use `unstructured.Unstructured` for `ServiceBinding` (https://github.com/openshift/odo/pull/4613) instead of using `ServiceBiding` struct types.

After #4648 is merged can move back to using `ServiceBinding` type.
This will make it easier to manage data in `ServiecBinding` CR and later it will be useful when implementing https://github.com/openshift/odo/issues/4543

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this?**

/kind code-refactoring

**What does this PR do / why we need it**:

#4648 changed client-go version, this PR will use ServiceBinding structure definition from ServiceBinding Operator instead of Unstructured which were used because of previous client-go versions incompatibility

**Which issue(s) this PR fixes**:

Fixes #4690 

**PR acceptance criteria**:

- [x] Unit test 

- [x] Integration test 

- [x] Documentation 

- [x] Update changelog

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

This branch is based on the branch kadel:ssa from #4648, so the firsts commits are from #4648, only consider the latest commits

### CODE REVIEW

File: pkg/odo/cli/component/common_link.go [hunk 1]
```diff
@@ -301,9 +295,18 @@ func (o *commonLinkOptions) run() (err error) {
 			return
 		}
 
+		// convert service binding request into a ma[string]interface{} type so
```
Comment (mik-dass) [inline pkg/odo/cli/component/common_link.go:298]: `map[string]`

File: pkg/odo/cli/component/common_link.go [hunk 2]
```diff
@@ -301,9 +295,18 @@ func (o *commonLinkOptions) run() (err error) {
...
+		// as to use it with dynamic client
+		serviceBindingMap := make(map[string]interface{})
+		intermediate, _ := json.Marshal(o.serviceBinding)
```
Comment (mik-dass) [inline pkg/odo/cli/component/common_link.go:301]: Why are we ignoring the error here?

Comment (feloy) [reply pkg/odo/cli/component/common_link.go]: Good catch, fixed

File: pkg/odo/cli/component/common_link.go (hunk 1 above)
Comment (feloy) [reply pkg/odo/cli/component/common_link.go]: thanks

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**